x/nameservice/keeper.go

```diff
@@ -46,6 +46,15 @@ func (k Keeper) SetWhois(ctx sdk.Context, name string, whois Whois) {
 	store.Set([]byte(name), k.cdc.MustMarshalBinaryBare(whois))
 }
 
+// Deletes the entire Whois metadata struct for a name
+func (k Keeper) DeleteWhois(ctx sdk.Context, name string) {
```

Comment:
How is this different that `DeleteName`? Why not just have a single method `DeleteWhois` or `DeleteName`. We should also return an error if the name doesn't exist.

Reply:
You're right. There isn't any difference between the two methods. Actually, Getter and Setter methods were defined in the form of SetWhoIs and GetWhoIs. And then additional methods were added for getting specific parameters from the store based on the name, such as: SetOwner(), GetOwner() etc. I thought of maintaining consistency by following the same structure. But yeah in the case of deleting the name, there aren't any additional methods that can be specified. Will keep `DeleteWhoIs`. And yeah I'll return an error if the name doesn't exist.